lib/src/transformer/impl_generation.dart

```diff
@@ -681,3 +735,17 @@ class ImplGenerator {
 }
 
 enum AccessorType {props, state}
+
+// The public Props|State|AbstractProps|AbstractState class signatures includes a with
+// <PropsClass>AccessorsMixin clause for dart 2 builder compatibility. But in Dart 1,
+// the transformer is able to generate the concrete accessors inline without a separate
+// mixin. For this reason, the transformer removes this with clause and then generates
+// the concrete accessors. To prevent the with clause being removed unnecessarily a check
+// is preformed to identify if the class has an annotation since the public class added
+// for dart 2 builder compatibility will not be annotated.
+void removeWithClauseIfNecessary(ClassDeclaration declaration, SourceFile sourceFile, TransformedSourceFile transformedFile) {
+  if (declaration == null) return;
+  if (declaration.metadata.isEmpty && declaration.withClause != null) {
```

Comment:
I missed this, we actually don't need the `metadata` check or the `withClause` anymore since we're capturing both the consumer-defined node as well as the companion node. In all of the usages of this method, we only ever pass the companion node so we can safely remove the with clause as long as the node itself isn't null